tests/integration/test.attachments.js

```diff
@@ -3256,6 +3256,38 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('Test rev purge with attachment', function () {
+      const db = new PouchDB(dbs.name);
+
+      if (typeof db._purge === 'undefined') {
+        console.log('purge is not implemented for adapter', db.adapter);
+        return;
+      }
+
+      const doc = { _id: 'foo' };
+      const base64 =
+        'iVBORw0KGgoAAAANSUhEUgAAAhgAAAJLCAYAAAClnu9J' +
+        'AAAgAElEQVR4Xuy9B7ylZXUu/p62T5nOMAPM0BVJICQi' +
+        'ogjEJN5ohEgQ';
+
+      return db.put(doc).then(function (res) {
+        return db.putAttachment('foo', 'foo.bin', res.rev, base64, 'image/png');
```

Comment:
I guess you are not a big fan of arrow functions or the `async/await` syntax but these would help the readability of this (and the following tests) a lot.

E.g. arrow function:
```javascript
return db.put(doc)
   .then((res) => db.putAttachment('foo', 'foo.bin', res.rev, base64, 'image/png'))
   .then(() => db.get(doc._id))
   .then((_doc) => db.purge(doc._id, _doc._rev))
   .then(...);
```
or with `async/await`
```javascript
const res = await db.put(doc);
await db.putAttachment('foo', 'foo.bin', res.rev, base64, 'image/png');
const _doc = await db.get(doc._id);  // this step can also be skipped because you can use the `res.rev` in the next line again
await db.purge(doc._id, _doc._rev);
...
```

Reply:
We’ll start introducing ES6 syntax soonish :)